src/skydb.ts

```diff
@@ -49,7 +49,7 @@ export async function setJSON(
   if (!revision) {
     // fetch the current value to find out the revision.
     console.log(3);
-    const entry = await this.registry.getEntry(publicKey, dataKey);
+    const entry = await promiseTimeout(this.registry.getEntry(publicKey, dataKey), 5000);
```

Comment:
we shouldn't add an arbitrary timeout here, we should rely on siad either 404 or 200